apps/andi/lib/andi/input_schemas/ingestion.ex

```diff
@@ -58,32 +58,75 @@ defmodule Andi.InputSchemas.Ingestion do
 
   def changeset(changes), do: changeset(%__MODULE__{}, changes)
 
-  def changeset(ingestion, changes) do
+  def changeset(%Andi.InputSchemas.Ingestion{} = ingestion, changes) do
     changes_with_id = StructTools.ensure_id(ingestion, changes)
     source_format = Map.get(changes, :sourceFormat, nil)
 
     ingestion
-    |> cast(changes_with_id, @cast_fields, empty_values: [])
-    |> validate_required(@required_fields, message: "is required")
-    |> cast_assoc(:schema, with: &DataDictionary.changeset(&1, &2, source_format), invalid_message: "is required")
-    |> cast_assoc(:extractSteps, with: &ExtractStep.changeset/2)
-    |> cast_assoc(:transformations, with: &Transformation.changeset/2)
-    |> foreign_key_constraint(:targetDataset)
+    |> Changeset.cast(changes_with_id, @cast_fields, empty_values: [""])
```

Comment:
Fixes a bug in product that prevents removing a dataset. (Appears to remove, but if you save and return, it restores the previous dataset)

Explanation: Because dataset is a required foreign key, the changeset rejects "" (empty string) as a valid change and falls back to the existing data. However, the changeset allows nil to be a valid change, effectively unsetting the field.

This line adds "" as an empty value, converting it to nil, therefore allowing the metadata form to unset the changeset field.